src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -835,6 +837,49 @@ export class AttachToDropdown extends SelectBox {
 	}
 }
 
+async function openNewNotebook(
+	telemetryService: IAdsTelemetryService,
+	notebookService: INotebookService,
+	configurationService: IConfigurationService,
+	commandService: ICommandService,
+	profile?: azdata.IConnectionProfile
+): Promise<void> {
+	telemetryService.createActionEvent(TelemetryKeys.TelemetryView.Notebook, TelemetryKeys.NbTelemetryAction.NewNotebookFromConnections)
+		.withConnectionInfo(profile)
+		.send();
+
+	const usePreviewFeatures = configurationService.getValue(CONFIG_WORKBENCH_ENABLEPREVIEWFEATURES);
+	const useVSCodeNotebooks = configurationService.getValue(CONFIG_WORKBENCH_USEVSCODENOTEBOOKS);
+	if (usePreviewFeatures && useVSCodeNotebooks) {
+		await commandService.executeCommand('ipynb.newUntitledIpynb');
```

Comment:
Could you instead put this logic into openNotebook?

Reply:
I could, but I wanted to maintain separation between the two implementations so that we don't negatively impact things that haven't been onboarded to the vscode experience yet. There's also the problem of openNotebook returning an editor pane. The newUntitledIpynb command lives in the bultin ipynb extension, so we woiuldn't get editor details back immediately like we do with openNotebook, which adds extra complication for whatever calls openNotebook.